static/js/libs/events.js

```diff
@@ -0,0 +1,64 @@
+const Events = {
+  constructor: function() {
+    this.events = {};
+    this.availableHandles = [];
+
+    return this;
+  },
+
+  _addListener: function(type, bindWindow) {
+    const bindTarget = bindWindow ? window : document.body;
+    bindTarget.addEventListener(type, this._handleEvent.bind(this, type));
+  },
+
+  _handleEvent: function(type, event) {
+    const eventTarget = event.target;
+
+    this.events[type].forEach(ev => {
+      // A bit hacky but this allows using the same api to bind to window for key, resize and scroll events
+      const target = ev.selector === 'window' ? window : eventTarget.closest(ev.selector);
```

Comment:
Yep, using 'window' as string selector seems a bit hacky. Would be better if selector could be either string (used as a selector) or a DOM object (like window, or just a DOM element) to be used directly.

Reply:
Updated this to check for type. Not 100% safe, but better then this hack!